syntax = "proto3";
import "google/protobuf/empty.proto";
option go_package = ".;proto";


service Order {
  //购物车
  rpc CartItemList(UserInfo) returns (CartItemListResponse);//获取用户所有购物车信息
  rpc DeleteCartItem(CartItemRequest) returns (google.protobuf.Empty);//删除购物车商品
  rpc CreateCartItem(CartItemRequest) returns (ShopCartItemInfoResponse);//创建购物车商品
  rpc UpdateCartItem(CartItemRequest) returns (google.protobuf.Empty);//更新购物车订单信息


  // 订单
  rpc CreateOrder(OrderRequest) returns (OrderInfoResponse);//新建订单
  rpc OrderList(OrderFilterRequest) returns (OrderListResponse);//订单列表
  rpc OrderDetail(OrderRequest) returns (OrderInfoDetailResponse);//订单详情
  rpc UpdateOrder(OrderStatus) returns (google.protobuf.Empty);//更新订单状态
}

message UserInfo {
  int32 id = 1;
}

message ShopCartItemInfoResponse {
  int32 id = 1;
  int32 userId = 2;
  int32 goodsId = 3;
  int32 nums = 4;
  bool checked = 5;
}
message CartItemListResponse{
  int32 total = 1;
  repeated ShopCartItemInfoResponse data = 2;
}
message CartItemRequest {
  int32 userId = 1;
  int32 goodsId = 2;
  int32 nums = 3;
  bool checked = 4;
}
message OrderRequest {
  int32 id = 1;
  int32 userId = 2;
  string address = 3;
  string phone = 4;
  string name = 5;
  string post = 6;
}
message OrderInfoResponse {
  int32 id = 1;
  int32 userId = 2;
  string orderSn = 3;
  string payType = 4;
  string status = 5;
  float amount = 6;
  string post = 7;
  float total = 8;
  string address = 9;
  string name = 10;
  string phone = 11;
}
message OrderFilterRequest {
  int32 userId = 1;
  int32 pages = 2;
  int32 pagePerNums = 3;
}
message OrderListResponse {
    float total = 1;
    repeated OrderInfoResponse data = 2;
}
message OrderItemResponse {
  int32 id = 1;
  int32 orderId = 2;
  int32 goodsId = 3;
  string goodsName = 4;
  string goodsImage = 5;
  float goodsPrice = 6;
  int32 nums = 7;
}
message OrderInfoDetailResponse {
  OrderInfoResponse orderInfo = 1;
  repeated OrderItemResponse data = 2;
}
message OrderStatus {
 string orderSn = 1;
 string status = 2;
}